fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/rules/PredicatePushDownRule.java

```diff
@@ -0,0 +1,336 @@
+        // be pushed into/through. Any that cannot accept the predicate will be skipped over and excluded from the
+        // rewritten expression.
+        //
+        final var belowExpressions = bindings.get(belowExpressionsMatcher);
+        for (final var belowExpression : belowExpressions) {
+            pushToVisitor.visit(belowExpression).ifPresent(newBelowExpressions::add);
+        }
+
+        if (newBelowExpressions.isEmpty()) {
+            //
+            // We were unable to push the predicates down into the child quantifier.
+            // Return without yielding a new expression.
+            //
+            return;
+        }
+
+        final var newRangesOverReference =
+                call.memoizeReference(Reference.from(newBelowExpressions));
+
+        final var newPushQuantifier = Quantifier.forEachBuilder()
+                .withAlias(pushQuantifier.getAlias())
+                .build(newRangesOverReference);
+
+        final var newOwnedQuantifiers = selectExpression.getQuantifiers()
+                .stream()
+                .map(quantifier -> quantifier.getAlias().equals(pushQuantifier.getAlias()) ? newPushQuantifier : quantifier)
+                .collect(ImmutableList.toImmutableList());
+
+        final var newSelectExpression = new SelectExpression(selectExpression.getResultValue(),
+                newOwnedQuantifiers,
+                ImmutableList.copyOf(fixedPredicates));
+
+        call.yieldExpression(newSelectExpression);
+    }
+
+    private static class PushToVisitor implements RelationalExpressionVisitorWithDefaults<Optional<? extends RelationalExpression>> {
+        @Nonnull
+        private final Set<? extends QueryPredicate> originalPredicates;
+        @Nonnull
+        private final Quantifier.ForEach pushQuantifier;
+
+        public PushToVisitor(@Nonnull final Set<? extends QueryPredicate> originalPredicates,
+                             @Nonnull final Quantifier.ForEach pushQuantifier) {
+            this.originalPredicates = originalPredicates;
+            this.pushQuantifier = pushQuantifier;
+        }
+
+        @Nonnull
+        private Set<? extends QueryPredicate> getOriginalPredicates() {
+            return originalPredicates;
+        }
+
+        @Nonnull
+        private Quantifier.ForEach getPushQuantifier() {
+            return pushQuantifier;
+        }
+
+        @Nonnull
+        private List<QueryPredicate> updatedPredicates(@Nonnull TranslationMap translationMap, @Nonnull Collection<? extends QueryPredicate> preExistingPredicates) {
+            var predicatesBuilder = ImmutableList.<QueryPredicate>builderWithExpectedSize(getOriginalPredicates().size() + preExistingPredicates.size())
+                    .addAll(preExistingPredicates);
+            for (QueryPredicate originalPredicate : getOriginalPredicates()) {
+                predicatesBuilder.add(originalPredicate.translateValues(translationMap));
```

Comment:
I think this needs to set `shoudSimplifyValues=true`

Reply:
Perhaps it was bad form, but that was actually implicit `translateValues(TranslationMap)`. I've removed that and replaced this call with a call to `translateCorrelations(translationMap, true)